activation/post.go

```diff
@@ -82,12 +83,57 @@ type PostSetupOpts struct {
 	DataDir          string              `mapstructure:"smeshing-opts-datadir"`
 	NumUnits         uint32              `mapstructure:"smeshing-opts-numunits"`
 	MaxFileSize      uint64              `mapstructure:"smeshing-opts-maxfilesize"`
-	ProviderID       int                 `mapstructure:"smeshing-opts-provider"`
+	ProviderID       PostProviderID      `mapstructure:"smeshing-opts-provider"`
 	Throttle         bool                `mapstructure:"smeshing-opts-throttle"`
 	Scrypt           config.ScryptParams `mapstructure:"smeshing-opts-scrypt"`
 	ComputeBatchSize uint64              `mapstructure:"smeshing-opts-compute-batch-size"`
 }
 
+type PostProviderID struct {
+	value *int64
+}
+
+// String implements pflag.Value.String.
+func (id *PostProviderID) String() string {
+	if id.value == nil {
+		return ""
+	}
+	return fmt.Sprintf("%d", *id.value)
+}
+
+// Type implements pflag.Value.Type.
+func (PostProviderID) Type() string {
+	return "PostProviderID"
+}
+
+// Set implements pflag.Value.Set.
+func (id *PostProviderID) Set(value string) error {
+	if len(value) == 0 {
+		id.value = nil
+		return nil
+	}
+
+	i, err := strconv.ParseInt(value, 10, 33)
+	if err != nil {
+		return fmt.Errorf("failed to parse PoST Provider ID (\"%s\"): %w", value, err)
+	}
+
+	id.value = new(int64)
+	*id.value = int64(i)
+	return nil
+}
+
+// SetInt64 sets the value of the PostProviderID to the given int64.
+func (id *PostProviderID) SetInt64(value int64) {
+	id.value = new(int64)
+	*id.value = value
```

Comment:
Wouldn't this work:
```suggestion
	id.value = &value
```

Reply:
In principle yes, but consider the following code:

```go
value := int64(5)
id := new(ProstProviderID)

id.SetInt64(value)

value = int64(2)
```

with the unchanged code `id` doesn't change when value is changed. If we use the reference of the passed in `value` `id` changes when `value` changes. Which would at least to me be unexpected behaviour.